napari/settings/_constants.py

```diff
@@ -42,3 +42,16 @@ class BrushSizeOnMouseModifiers(StrEnum):
     CTRL_ALT = 'Control+Alt'
     CTRL_SHIFT = 'Control+Shift'
     DISABLED = 'Disabled'  # a non-existent modifier that is never activated
+
+
+class PluginShimWarningLevel(StrEnum):
+    """Warning level for shimmed plugins.
+
+    PluginShimWarningLevel.ALWAYS
+        Warnings will be shown for all shimmed plugins on each startup.
+    PluginShimWarningLevel.NEW
+        Warnings will be shown when a shimmed plugin is newly installed.
+    """
+
+    ALWAYS = 'always'
```

Comment:
Now that there's only two levels of warning, this could actually be a boolean "Only warn on new adapted plugins". Happy to change if we think that's better

Reply:
Yes, I would simplify to a boolean.